dali/operators/reader/fits_reader_op.cc

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2020-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "dali/core/backend_tags.h"
+#include "dali/core/static_switch.h"
+#include "dali/kernels/slice/slice_cpu.h"
+#include "dali/kernels/slice/slice_flip_normalize_permute_pad_cpu.h"
+#include "dali/kernels/transpose/transpose.h"
+#include "dali/operators/reader/fits_reader_op.h"
+#include "dali/pipeline/data/backend.h"
+
+namespace dali {
+namespace detail {
+inline int FitsReaderOutputFn(const OpSpec &spec) {
+  // there is default value provided, so no checking if arg exists is needed
+  return spec.GetRepeatedArgument<int>("hdu_indices").size();
+}
+
+}  // namespace detail
+
+DALI_REGISTER_OPERATOR(experimental__readers__Fits, FitsReaderCPU, CPU);
+
+DALI_SCHEMA(experimental__readers__Fits)
+    .DocStr(R"(Reads Fits image HDUs from a directory.
+
+This operator can be used in the following modes:
+
+1. Read all files from a directory indicated by ``file_root`` that match given ``file_filter``.
+2. Read file names from a text file indicated in ``file_list`` argument.
+3. Read files listed in ``files`` argument.
+4. Number of outputs per sample corresponds to the length of ``hdu_indices`` argument. By default,
+first HDU with data is read from each file, so the number of outputs defaults to 1. 
```

Comment:
Nitpick, can be done as a follow-up

```suggestion

The number of outputs per sample corresponds to the length of ``hdu_indices`` argument. By default,
first HDU with data is read from each file, so the number of outputs defaults to 1. 
```